### DISCUSSION TITLE
build docs from main repo

### DISCUSSION THREAD

**Discussion by chr15m** [piku/piku#370] [Ideas]
Often changes to Piku require corresponding documentation updates. In recent PRs this has meant people have to create two separate PRs against both the main repo and also the documentation repo. This is extra work and introduces coordination issues.

I wonder if it would be better to move the sources for docs into the main repo and have a GH action build the documentation site from that, monorepo style?

I think this would simplify things for contributors and maintainers. I am happy to do the work but I wanted to ask first.

**Comment by rcarmo** [comment]
Well, I'm not sure how to get mkdocs (and gh-pages) to work in that kind of scenario--I went for the simplest possible approach, since we weren't getting a substantial amount of contributions before... I've also been meaning to completely excise the examples and moving them to separate repos, but I just don't have the bandwidth. Go ahead and give it a go...

**Reply by chr15m** [reply]
I'll try to find some time to move those remaining examples out, and also I think we can remove the experimental `binary` folder as that was a WIP and I don't think anybody is using it.

**Comment by jfmatth** [comment]
The webhook for one of the other could probably fire the web hook for the
other one maybe that's how we get one repo update to update another repel

I hope that made sense

**Reply by rcarmo** [reply]
I think that's how other projects do it - I came across a Python project that also used mkdocs in the main repo the other day, but I forgot which :(
